liblttng-ust-java-agent/java/lttng-ust-agent-jul/org/lttng/ust/agent/jul/LttngLogHandler.java

```diff
@@ -84,11 +88,30 @@ public LttngLogHandler() throws IOException, SecurityException {
 			throw new IOException(e);
 		}
 
+		/* Initialize handler specific properties */
+		LogManager manager = LogManager.getLogManager();
+		String cname = getClass().getName();
+		logSourceMethod = (getBooleanProperty(manager.getProperty(cname + ".logSourceMethod"), true));
+		logSourceClass = (getBooleanProperty(manager.getProperty(cname + ".logSourceClass"), true));
+
 		/** Register to the relevant agent */
 		agent = LttngJulAgent.getInstance();
 		agent.registerHandler(this);
 	}
 
+	private boolean getBooleanProperty(String val, boolean defaultValue) {
+	        if (val == null) {
```

Comment:
Btw, I have a "Java dev" branch of UST, feel free to use it: 
https://github.com/alexmonthy/lttng-ust/commit/2adb17e936d9c8284edb99037e95a5a25dfa8f65
https://github.com/alexmonthy/lttng-ust/commit/1a5149bd2856c139f62b0aadef7649d05b9af365